app/models/post.rb

```diff
@@ -1,2 +1,16 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes, class_name: 'Like'
+  has_many :comments, class_name: 'Comment'
+
+  after_create :update_users_count
+
+  def update_users_count
+    author = User.find(author_id)
+    author.increment(:posts_counter)
+  end
```

Comment:
- [x] Please let's correct the following in this method:

 1. The name: is it users count or posts count that we are implementing? It should be posts count, right? Please update the name of the method to `update_posts_count`.
 2. The next line for finding the author is not necessary. A post already belongs to an author and its model can directly access the author object and modify its properties. Please consider working with the suggestion below:
 
 ```suggestion
  def update_posts_count
    author.increment!(:posts_counter)
  end
  ```
  
  Lastly, please move this method to the bottom of the class and make it private just as we've done with others up there.

Reply:
This issue is well resolved! ✔️